src/platform/stm32/libs/libusbhost/src/usbh_core.c

```diff
@@ -134,18 +134,19 @@ static void device_register(void *descriptors, uint16_t descriptors_len, usbh_de
 		switch (desc_type) {
 		case USB_DT_INTERFACE:
 		{
-			LOG_PRINTF("INTERFACE_DESCRIPTOR\n");
 			struct usb_interface_descriptor *iface = (void*)&buf[i];
 			device_info.ifaceClass = iface->bInterfaceClass;
 			device_info.ifaceSubClass = iface->bInterfaceSubClass;
 			device_info.ifaceProtocol = iface->bInterfaceProtocol;
+			LOG_PRINTF("INTERFACE_DESCRIPTOR  Cls %02X  Sub %02X  Prot %02X \n", iface->bInterfaceClass, iface->bInterfaceSubClass, iface->bInterfaceProtocol);
 			if (find_driver(dev, &device_info)) {
+				uint8_t *ibuf = (uint8_t *)iface;
 				int k = 0;
 				while (k < descriptors_len) {
```

Comment:
While before we always used data of the first interface, now we don't check for end of buffer correctly anymore, right? Should probably be `while (k < descriptors_len - i)` now.

Reply:
right, -i is missing, adding it